frontend/src/pages/ProgressionChecker/TableView/TableView.jsx

```diff
@@ -1,53 +1,148 @@
+                    termTaken: plannedCourses[courseCode].termTaken,
+                  });
+                  newTableLayout[group][subgroup].sort(
+                    (a, b) => a.termTaken.localeCompare(b.termTaken),
+                  );
+                }
+              });
+            } else {
+              // If there is no specified course list for the subgroup, then manually
+              // show the added courses for the subgroup.
+              Object.keys(plannedCourses).forEach((courseCode) => {
+                const courseData = plannedCourses[courseCode];
+                if (courseData && courseData.type === subgroup) {
+                  newTableLayout[group][subgroup].push({
+                    key: courseCode,
+                    title: plannedCourses.courseCode.title,
+                    UOC: plannedCourses.courseCode.UOC,
+                    termTaken: plannedCourses.courseCode.termTaken,
+                  });
+                }
+              });
+            }
+          }
+        });
+        if (structure[major][group].name) {
+          // Append structure[major] group name if exists
+          const newGroup = `${group} - ${structure[major][group].name}`;
+          newTableLayout[newGroup] = newTableLayout[group];
+          delete newTableLayout[group];
+        }
+      });
+    });
+    setTableLayout(newTableLayout);
+  };
+
+  useEffect(() => {
+    getPastCourses();
+    generateTableStructure(pastCourses);
+  }, [isLoading, structure, years, startYear, courses]);
+
   const columns = [
+    {
+      title: "Course Code",
+      dataIndex: "key",
+      key: "key",
+    },
     {
       title: "Course Name",
-      dataIndex: "name",
-      key: "name",
-      render: (name) => name,
+      dataIndex: "title",
+      key: "title",
     },
     {
       title: "UOC",
       dataIndex: "UOC",
       key: "UOC",
     },
     {
-      title: "Faculty",
-      dataIndex: "faculty",
-      key: "faculty",
-      filters: [
-        {
-          text: "UNSW Business School",
-          value: "UNSW Business School",
-        },
-        {
-          text: "School of Computer Science and Engineering",
-          value: "School of Computer Science and Engineering",
-        },
-      ],
-      onFilter: (value, record) => record.faculty.indexOf(value) === 0,
-    },
-    {
-      title: "State",
-      dataIndex: "state",
-      key: "state",
-      sorter: (a, b) => a.state.localeCompare(b.state),
-    },
-    {
-      title: "Time",
-      dataIndex: "time",
-      key: "time",
-      sorter: (a, b) => a.time.localeCompare(b.time),
+      title: "Term Taken",
+      dataIndex: "termTaken",
+      key: "termTaken",
     },
   ];
 
   return (
-    <div className="listPage">
-      <Table className="table-striped-rows" dataSource={checkercourses.corecourses} columns={columns} />
+    <div className="tableViewContainer">
+      {isLoading ? (
+        <br />
```

Comment:
Add a TODO comment to implement a loading skeleton later on as a future ticket feature